src/Microsoft.Repl/Input/InputManager.cs

```diff
@@ -18,6 +18,24 @@ public class InputManager : IInputManager
 
         public bool IsOverwriteMode { get; set; }
 
+        public int CaretPosition { get; private set; }
+
+        public void MoveCaret(int positions)
+        {
+            if (CaretPosition + positions < 0)
+            {
+                CaretPosition = 0;
```

Comment:
It certainly _shouldn't_ if all the other calculations were right. But I put the checks in there to try to prevent crashes in case it did. This way, if something did go wrong, you might end up with unexpected values in the input buffer (bad) but they'd correct as soon as you hit enter and started a new command. The crash would cause you to lose all your session state.